nextcord/guild.py

```diff
@@ -3457,3 +3461,171 @@ async def register_application_commands(self, *commands: BaseApplicationCommand)
+
+        Returns
+        --------
+        :class:`AutoModerationRule`
+        """
+        rule = await self._state.http.get_automod_rule(guild_id=self.id, rule_id=rule_id)
+        return AutoModerationRule(guild=self, state=self._state, data=rule)
+
+    def get_automod_rule(self, rule_id: int) -> Optional[AutoModerationRule]:
+        return self._state.get_automod_rule(id=rule_id)
+
+    @overload
+    async def create_automod_rule(
+        self,
+        *,
+        name: str,
+        event_type: EventType = EventType.message_send,
+        trigger_type: TriggerType,
+        keyword_filters: Optional[List[str]] = MISSING,
+        notify_channel: Optional[GuildChannel] = MISSING,
+        enabled: Optional[bool] = True,
+        exempt_roles: Optional[List[Role]] = MISSING,
+        exempt_channels: Optional[List[GuildChannel]] = MISSING,
+    ) -> AutoModerationRule:
+        ...
+
+    @overload
+    async def create_automod_rule(
+        self,
+        *,
+        name: str,
+        event_type: EventType = EventType.message_send,
+        trigger_type: TriggerType,
+        keyword_filters: Optional[List[str]] = MISSING,
+        timeout_seconds: Optional[int] = MISSING,
+        enabled: Optional[bool] = True,
+        exempt_roles: Optional[List[Role]] = MISSING,
+        exempt_channels: Optional[List[GuildChannel]] = MISSING,
+    ) -> AutoModerationRule:
+        ...
+
+    @overload
+    async def create_automod_rule(
+        self,
+        *,
+        name: str,
+        event_type: EventType = EventType.message_send,
+        trigger_type: TriggerType,
+        presets: Optional[KeywordPresetType] = MISSING,
+        notify_channel: Optional[GuildChannel] = MISSING,
+        enabled: Optional[bool] = True,
+        exempt_roles: Optional[List[Role]] = MISSING,
+        exempt_channels: Optional[List[GuildChannel]] = MISSING,
+    ) -> AutoModerationRule:
+        ...
+
+    @overload
+    async def create_automod_rule(
+        self,
+        *,
+        name: str,
+        event_type: EventType = EventType.message_send,
+        trigger_type: TriggerType,
+        presets: Optional[KeywordPresetType] = MISSING,
+        notify_channel: Optional[GuildChannel] = MISSING,
+        enabled: Optional[bool] = True,
+        exempt_channels: Optional[List[GuildChannel]] = MISSING,
+    ) -> AutoModerationRule:
+        ...
+
+    async def create_automod_rule(
+        self,
+        *,
+        name: str,
+        event_type: EventType = EventType.message_send,
+        trigger_type: TriggerType,
+        keyword_filters: Optional[List[str]] = MISSING,
+        presets: Optional[KeywordPresetType] = MISSING,
+        notify_channel: Optional[GuildChannel] = MISSING,
+        timeout_seconds: Optional[int] = MISSING,
+        enabled: Optional[bool] = True,
+        exempt_roles: Optional[List[Role]] = MISSING,
+        exempt_channels: Optional[List[GuildChannel]] = MISSING,
+    ) -> AutoModerationRule:
```

Comment:
Optional denotes `T | None`, `MISSING` is not `None` so these should not be marked as optional
```suggestion
    @overload
    async def create_automod_rule(
        self,
        *,
        name: str,
        event_type: EventType = EventType.message_send,
        trigger_type: TriggerType,
        keyword_filters: List[str] = MISSING,
        notify_channel: GuildChannel = MISSING,
        enabled: bool = True,
        exempt_roles: List[Role] = MISSING,
        exempt_channels: List[GuildChannel] = MISSING,
    ) -> AutoModerationRule:
        ...

    @overload
    async def create_automod_rule(
        self,
        *,
        name: str,
        event_type: EventType = EventType.message_send,
        trigger_type: TriggerType,
        keyword_filters: List[str] = MISSING,
        timeout_seconds: int = MISSING,
        enabled: bool = True,
        exempt_roles: List[Role] = MISSING,
        exempt_channels: List[GuildChannel] = MISSING,
    ) -> AutoModerationRule:
        ...

    @overload
    async def create_automod_rule(
        self,
        *,
        name: str,
        event_type: EventType = EventType.message_send,
        trigger_type: TriggerType,
        presets: KeywordPresetType = MISSING,
        notify_channel: GuildChannel = MISSING,
        enabled: bool = True,
        exempt_roles: List[Role] = MISSING,
        exempt_channels: List[GuildChannel] = MISSING,
    ) -> AutoModerationRule:
        ...

    @overload
    async def create_automod_rule(
        self,
        *,
        name: str,
        event_type: EventType = EventType.message_send,
        trigger_type: TriggerType,
        presets: KeywordPresetType = MISSING,
        notify_channel: GuildChannel = MISSING,
        enabled: bool = True,
        exempt_channels: List[GuildChannel] = MISSING,
    ) -> AutoModerationRule:
        ...

    async def create_automod_rule(
        self,
        *,
        name: str,
        event_type: EventType = EventType.message_send,
        trigger_type: TriggerType,
        keyword_filters: List[str] = MISSING,
        presets: KeywordPresetType = MISSING,
        notify_channel: GuildChannel = MISSING,
        timeout_seconds: int = MISSING,
        enabled: bool = True,
        exempt_roles: List[Role] = MISSING,
        exempt_channels: List[GuildChannel] = MISSING,
    ) -> AutoModerationRule:
```